packages/formik-material-ui-pickers/src/errorHandler.ts

```diff
@@ -0,0 +1,11 @@
+import { ReactNode } from 'react';
+
+export function createErrorHandler(fieldError: unknown, fieldName: string, setFieldError: (field: string, message: string) => void) {
+  return (error : ReactNode) => {
+    if (error !== fieldError && error !== '') {
```

Comment:
This is where the meaningful change is: I removed the `&& !fieldError` from condition. In cases where there was a validation error coming from Formik, `error` is empty and `fieldError` has value. Then `setFieldError` would incorrectly fire and set the field error to `undefined` (effectively breaking the validation).